src/main/java/com/puppycrawl/tools/checkstyle/filters/XpathFilterElement.java

```diff
@@ -277,7 +267,8 @@ public boolean equals(Object other) {
                 && Objects.equals(checkPattern, xpathFilter.checkPattern)
                 && Objects.equals(messagePattern, xpathFilter.messagePattern)
                 && Objects.equals(moduleId, xpathFilter.moduleId)
-                && Objects.equals(xpathQuery, xpathFilter.xpathQuery);
+                && Objects.equals(xpathQuery, xpathFilter.xpathQuery)
+                && Objects.equals(allPropertiesAreNull, xpathFilter.allPropertiesAreNull);
```

Comment:
There is no reason to put this field in hash and equals method. If checks above for other fields were true, this one wont be different. 
I mean that there is no case when this comparison will be not true when we reached it.

Reply:
Done